pymatgen/io/lammps/inputs.py

```diff
@@ -14,18 +17,682 @@
+        If a stage name is specified, it will be replaced or set only for this stage.
+        If the command is set multiple times in the file/stage, it will be replaced based on "how":
+        either the first occurrence, all of them, or the index of the occurrence.
+
+        Args:
+            command (str): String representing the command to change, e.g., "units".
+            argument (str): String with the new value for the command, e.g., "atomic".
+            stage_name (str): String giving the stage name where the change should take place.
+            how (str or int or list): "all" for changing all occurrences of the command within the stage_name
+                or the whole input file, "first" for the first occurrence, int i for the i-th time the command
+                is present in the stage_name or the whole input file, starting at 0. Can be a list of indexes as well.
+        """
+        # Get the stages to look in
+        stages_to_look = [stage_name] if stage_name else self.stages_names
+
+        # Translates how into range of indices
+        if how == "first":
+            how = [0]
+        elif how == "all":
+            getargs = self.get_args(command, stage_name)
+            N = 1 if isinstance(getargs, str) else len(getargs)
+            how = [i for i in range(N)]
+        elif isinstance(how, int):
+            how = [how]
+
+        # Look for occurrences in the relevant stages
+        i = 0
+        for i_stage, (stage, cmd_list) in enumerate(self.list_of_commands):
+            if stage in stages_to_look:
+                for i_cmd, (cmd, _) in enumerate(cmd_list):
+                    if command == cmd:
+                        if i in how:
+                            self.list_of_commands[i_stage][1][i_cmd][1] = argument
+                        i += 1
+
+    def add_stage(
+        self,
+        command: str | list[str] | dict[str, str | float],
+        stage_name: str | None = None,
+        after_stage: str | None = None,
+    ):
+        r"""
+        Adds LAMMPS command(s) and its arguments to LAMMPS input file.
+
+        Examples:
+            1) In order to add a stage defining the potential to be used, you can use:
+            ```
+            your_input_file.add_stage(
+                command=["pair_coeff 1 1 morse 0.0580 3.987 3.404", "pair_coeff 1 4 morse 0.0408 1.399 3.204"],
+                stage_name="Definition of the potential"
+            )
+            ```
+
+            2) Another stage could consist in an energy minimization. In that case, the commands could look like
+            ```
+            command = [
+                "thermo 1",
+                "thermo_style custom step lx ly lz press pxx pyy pzz pe",
+                "dump dmp all atom 5 run.dump",
+                "min_style cg",
+                "fix 1 all box/relax iso 0.0 vmax 0.001",
+                "minimize 1.0e-16 1.0e-16 5000 10000",
+                "write_data run.data"
+            ]
```

Comment:
The example is correct. Using this with the add_stage, we get e.g.
```
In [19]: lmp_input = LammpsInputFile()
In [20]: lmp_input.add_stage(command = [
    ...:                 "thermo 1",
    ...:                 "thermo_style custom step lx ly lz press pxx pyy pzz pe",
    ...:                 "dump dmp all atom 5 run.dump",
    ...:                 "min_style cg",
    ...:                 "fix 1 all box/relax iso 0.0 vmax 0.001",
    ...:                 "minimize 1.0e-16 1.0e-16 5000 10000",
    ...:                 "write_data run.data"
    ...:             ])
In [21]: lmp_input.list_of_commands
Out[21]: 
[['Stage 1',
  [['thermo', '1'],
   ['thermo_style', 'custom step lx ly lz press pxx pyy pzz pe'],
   ['dump', 'dmp all atom 5 run.dump'],
   ['min_style', 'cg'],
   ['fix', '1 all box/relax iso 0.0 vmax 0.001'],
   ['minimize', '1.0e-16 1.0e-16 5000 10000'],
   ['write_data', 'run.data']]]]
```

I think there is a confusion between "command" in add_stage (and other places in the code) and "list_of_commands" (LammpsInputFile attribute) because it's not very clear in the documentation. Would it be better to rename list_of_commands? Any suggestion? It was initially input_settings, and we changed it for clarity but maybe it had the opposite effect...

Reply:
Yes, I see that you're right. As I said in the other comment, I agree that `list_of_commands` is a confusing name; I think it should be `.stages`.

Semantically, I also feel that whatever is passed to `add_stage` should have the same syntax as an element of `.stages` (that's what led to my confusion about the example - since `list_of_commands` takes `
list_of_commands = [  ["Stage 1", [(cmd1, args1), (cmd2, args2)] ]`, I was expecting the `command` here to have the same format.